### PR TITLE
feature: Enable visionOS Support

### PR DESCRIPTION
**Pull request by calvincestari**
Closes https://github.com/apollographql/apollo-ios/issues/3320.

With [SQLite.swift being updated to support visionOS](https://github.com/stephencelis/SQLite.swift/releases/tag/0.15.1) this:
* Updates our SQLite.swift dependency version
* Adds `visionOS` platform declarations
* Updates other dependency versions

### CODE REVIEW

File: apollo-ios-codegen/Package.swift [hunk 1]
```diff
@@ -19,10 +19,10 @@ let package = Package(
       .upToNextMajor(from: "1.0.0")),
     .package(
       url: "https://github.com/apple/swift-collections",
-      .upToNextMajor(from: "1.0.6")),
+      .upToNextMajor(from: "1.1.0")),
```
Comment (calvincestari) [inline apollo-ios-codegen/Package.swift:22]: I'm on the fence about whether we should include these here or not? I could easily be persuaded to revert this change.

**Review by AnthonyMDev — APPROVED** [review]
Excited to get this done!

**Comment by loganblevins** [comment]
Nice work!!
